pandas/core/ops.py

```diff
@@ -638,17 +639,33 @@ def wrapper(left, right, name=name, na_op=na_op):
                 if ridx is not None:
                     rvalues = algos.take_1d(rvalues, ridx)
 
-            arr = na_op(lvalues, rvalues)
-
```

Comment:
I would either do this directly in ``na_op`` (not sure if that's easy/possible) or 

```
def safe_na_op(lvalues, rvalues): # better name...
     try:
           return na_op(lvalues, rvalues)
     except Exception:
           if is_object_dtype(........):
```

then
```
result = wrrap_results(maybe_na_op(lvalues, rvalues))
```


```
# only need 1 return statement
return left._constructor(result, .....)
```